src/Membership/MembershipRequestHandler.php

```diff
@@ -113,13 +112,18 @@ public function terminateMembership(UnregistrationCommand $command, Adherent $ad
         $this->committeeFeedManager->removeAuthorItems($adherent);
         $this->activitySubscriptionManager->removeAdherentActivities($adherent);
 
-        if ($token) {
-            $this->manager->remove($token);
-        }
         if ($summary) {
             $this->manager->remove($summary);
         }
-        $this->manager->remove($adherent);
+
+        $adherent->leave();
+
+        if ($removeAccount) {
```

Comment:
if `removeAccount` is `false` we remove anyway all that
```
$this->removeAdherentMemberShips($adherent);
$this->citizenInitiativeManager->removeOrganizerCitizenInitiatives($adherent);
$this->eventManager->removeOrganizerEvents($adherent);
$this->registrationManager->anonymizeAdherentRegistrations($adherent);
$this->committeeFeedManager->removeAuthorItems($adherent);
$this->activitySubscriptionManager->removeAdherentActivities($adherent);
```
juste `$adherent->leave();` is't enough?